cmd/example/simple/simple.go

```diff
@@ -34,9 +40,10 @@ func main() {
 	lpms := core.New(&core.LPMSOpts{WorkDir: fmt.Sprintf("%v/.tmp", dir)})
 
 	lpms.HandleRTMPPublish(
-		func(url *url.URL) (strmID string) {
+		func(url *url.URL) stream.AppData {
 			glog.Infof("Stream has been started!: %v", url)
-			return randString(10)
+			s := exampleStream(randString(10))
+			return &s
```

Comment:
Please change this to:
```golang
return &exampleStream(randString(10))
```

Reply:
Actually, I don't think the `&` is needed at all in this case - you are returning an interface.  Please see if:
```
return exampleStream(randString(10))
```
works without the ampersand.   If it does need the ampersand, it can be returned straight away without an intermediate.

Reply:
To do the second way, the `StreamID` method needs to be updated to take a non-pointer receiver. Not ideal for most normal usage but is OK for this case.

```diff
diff --git a/cmd/example/simple/simple.go b/cmd/example/simple/simple.go
index 5ec53f2..4ac4099 100644
--- a/cmd/example/simple/simple.go
+++ b/cmd/example/simple/simple.go
@@ -15,8 +15,8 @@ import (
 
 type exampleStream string
 
-func (t *exampleStream) StreamID() string {
-	return string(*t)
+func (t exampleStream) StreamID() string {
+	return string(t)
 }
 
 func randString(n int) string {
@@ -42,8 +42,7 @@ func main() {
 	lpms.HandleRTMPPublish(
 		func(url *url.URL) stream.AppData {
 			glog.Infof("Stream has been started!: %v", url)
-			s := exampleStream(randString(10))
-			return &s
+			return exampleStream(randString(10))
 		},
 
 		func(url *url.URL, rs stream.RTMPVideoStream) (err error) {
```